bolt-modules/boltlib/lib/puppet/functions/puppetdb_query.rb

```diff
@@ -16,15 +18,35 @@
     param 'Variant[String, Array[Data]]', :query
     return_type 'Array[Data]'
   end
+
+  # rubocop:disable Layout/LineLength
+  # Make a query to a named PuppetDB instance.
+  #
+  # @param query A PQL query.
+  #   Learn more about [Puppet's query language](https://puppet.com/docs/puppetdb/latest/api/query/tutorial-pql.html), PQL.
+  # @param instance The PuppetDB instance to query.
+  # @return Results of the PuppetDB query.
+  # @example Request certnames for all nodes using a named PuppetDB instance
+  #   puppetdb_query('nodes[certname] {}', 'instance-1')
+  # rubocop:enable Layout/LineLength
+  dispatch :make_query_with_instance do
+    param 'Variant[String, Array[Data]]', :query
+    param 'String', :instance
+    return_type 'Array[Data]'
+  end
   # The query type could be more specific ASTQuery = Array[Variant[String, ASTQuery]]
 
   def make_query(query)
+    make_query_with_instance(query, nil)
+  end
+
+  def make_query_with_instance(query, instance)
     puppetdb_client = Puppet.lookup(:bolt_pdb_client)
     # Bolt executor not expected when invoked from apply block
     executor = Puppet.lookup(:bolt_executor) { nil }
     # Send Analytics Report
     executor&.report_function_call(self.class.name)
 
-    puppetdb_client.make_query(query)
+    puppetdb_client.make_query(query, instance)
```

Comment:
This change will cause trouble in PE where we only support the `make_query(query)` signature. We could either implement support for this new signature (with an error, since `instance` doesn't make sense in that context) or fork this function in PE and add the error there.